api/tests/unit/application/campaign/campaign-controller_test.js

```diff
@@ -0,0 +1,178 @@
+const { sinon, expect, factory } = require('../../../test-helper');
+
+const campaignController = require('../../../../lib/application/campaigns/campaign-controller');
+const usecases = require('../../../../lib/domain/usecases');
+const campaignRepository = require('../../../../lib/infrastructure/repositories/campaign-repository');
+const userRepository = require('../../../../lib/infrastructure/repositories/user-repository');
+const campaignSerializer = require('../../../../lib/infrastructure/serializers/jsonapi/campaign-serializer');
+const { UserNotAuthorizedToCreateCampaignError, EntityValidationError } = require('../../../../lib/domain/errors');
+
+describe('Unit | Application | Controller | Campaign', () => {
+
+  describe('#save', () => {
+
+    let replyStub;
+    let codeStub;
+
+    beforeEach(() => {
+      sinon.stub(usecases, 'createCampaign');
+      sinon.stub(campaignSerializer, 'deserialize');
+      sinon.stub(campaignSerializer, 'serialize');
+      codeStub = sinon.stub();
+      replyStub = sinon.stub().returns({
+        code: codeStub,
+      });
+    });
+
+    afterEach(() => {
+      usecases.createCampaign.restore();
+      campaignSerializer.deserialize.restore();
+      campaignSerializer.serialize.restore();
+    });
+
+    it('should call the use case to create the new campaign', () => {
+      // given
+      const userId = 1245;
+      const request = { auth: { credentials: { userId } } };
```

Comment:
Ajout d'un context 'when user has access to the organization' qui set l'userId et la request ?

Reply:
On présume pas que le user a le droit de la crée dans ce premier test. c'est le usecase qui va faire cette vérif. Je vais donner une autre valeur à userId histoire qu'on pense pas que les deux cas sont similaire. Le premier test vérifie que le controller appelle les bons truks quoiqu'il arrive, le deuxième vérifie le comportement si ça se passe bien.